### PR TITLE
V1.4.0

### PR DESCRIPTION
**Pull request by aaronlevy**
Main changes:
- Update vendored k8s code to v1.4.0+coreos.0
- Update manifests to use v1.4.0_coreos.0 hyperkube image

### CODE REVIEW

File: hack/single-node/user-data.sample [hunk 1]
```diff
@@ -32,12 +32,11 @@ coreos:
         ExecStartPre=/bin/mkdir -p /srv/kubernetes/manifests
         ExecStartPre=/bin/mkdir -p /etc/kubernetes/checkpoint-secrets
         ExecStart=/usr/lib/coreos/kubelet-wrapper \
-          --api-servers=https://172.17.4.100:443 \
           --kubeconfig=/etc/kubernetes/kubeconfig \
```
Comment (derekparker) [inline hack/single-node/user-data.sample]: Needs `--require-kubeconfig`

File: pkg/asset/internal/templates.go [hunk 2]
```diff
@@ -349,20 +359,25 @@ spec:
       - name: healthz
         image: gcr.io/google_containers/exechealthz-amd64:1.1
         resources:
+          # keep request = limit to keep this container in guaranteed class
           limits:
             cpu: 10m
```
Comment (pbx0) [inline pkg/asset/internal/templates.go]: nit: this is an slightly old version of kube-dns. A recent commit during beta removed a few of the cpu limits for kube-dns. See: https://github.com/kubernetes/kubernetes/commit/2b805e3d0f4bf9d377325e0433c799c3f4a56fb2

Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: Nice catch. Updated to reflect those changes.

File: hack/single-node/user-data.sample (hunk 1 above)
Comment (aaronlevy) [inline hack/single-node/user-data.sample]: Added

**Pull request merged**